core/debugger/remote_debugger_peer.cpp

```diff
@@ -192,11 +257,14 @@ Error RemoteDebuggerPeerTCP::connect_to_host(const String &p_host, uint16_t p_po
 
 void RemoteDebuggerPeerTCP::_thread_func(void *p_ud) {
 	// Update in time for 144hz monitors
+	// FIXME that precision is nonsense, we don't have that kind of resolution from sleeping (ms on Windows, scheduling intervals, etc.)
```

Comment:
Not sure if outdated but 6,9 ms is the expected frame rate for a 144Hz monitor.

Reply:
For this use case, I doubt it's really a problem if the actual sleep duration ends up being 6ms, 7ms, 10ms or anything in between.

Reply:
Yeah, maybe we should make this a define somewhere (e.g. `THREAD_POLL_DELAY`), we use this magic number in more than one place IIRC.

Reply:
I am sorry my comment is garbage :).  It is technically entirely correct but not at all clear.  Let me try again:

REVISIT:  Whoever designed this needs to look at this again.  The code makes a claim that sleeping for this delay would result in waking often enough for 144Hz updates.  However, this is very unlikely to be correct.  For example, a Windows default installation has a timer resolution of around 64Hz.  Also, there is the fact that a normal user mode thread in a non-real time OS may not even be able to get scheduled with close to that accuracy.   Therefore, it is very likely that whatever problem is supposed to be addressed by this delay value is actually not being handled.   Specifying it at this precision is very misleading when the underlying primitive is actually a user mode sleep at ms resolution.

```
void OS_Windows::delay_usec(uint32_t p_usec) const {
	if (p_usec < 1000) {
		Sleep(1);
	} else {
		Sleep(p_usec / 1000);
	}
}
```

Reply:
I don't know what problem is being solved, so I left this comment but never got back to it.  I should have written a better comment.   I don't know if anyone ever looks at REVISIT so maybe it is a FIXME if you folks agree.

Reply:
> For this use case, I doubt it's really a problem if the actual sleep duration ends up being 6ms, 7ms, 10ms or anything in between.

To be clear:  if I am correct and the timer resolution is 64Hz, then sleeping at all (any longer than just yielding the CPU) would result in maximum update rate of 64Hz, which clearly does not service 144 fps as intended.

Reply:
https://randomascii.wordpress.com/2020/10/04/windows-timer-resolution-the-great-rule-change/